xbmc/filesystem/CurlFile.cpp

```diff
@@ -694,6 +703,33 @@ void CCurlFile::ParseAndCorrectUrl(CURL &url2)
   std::string strProtocol = url2.GetTranslatedProtocol();
   url2.SetProtocol(strProtocol);
 
+  // lookup host in DNS cache
+  std::string resolvedHost;
+  if (CDNSNameCache::Lookup(url2.GetHostName(), resolvedHost))
+  {
+    struct curl_slist* tempCache;
+    int entryPort = url2.GetPort();
+
+    if (entryPort == 0)
+    {
+      if (url2.GetProtocol() == "http" || url2.GetProtocol() == "dav")
+        entryPort = 80;
+      else if (url2.GetProtocol() == "https" || url2.GetProtocol() == "davs")
+        entryPort = 443;
+      else if (url2.GetProtocol() == "ftp")
+        entryPort = 21;
+      else if (url2.GetProtocol() == "ftps")
```

Comment:
Many calls to `url2.GetProtocol()`here. Maybe define an local variable for the value? Even if `url2.GetProtocol()` is cheap and inline and what not today, you never know how implementation will change in the future, performance-wise.

Reply:
fixed